src/ClientTransactionTask.h

```diff
@@ -0,0 +1,264 @@
+        KeyHash keyHash;        // start key hash value
+
+        /**
+         * The operator < is overridden to implement the
+         * correct comparison for the CommitCache map.
+         */
+        bool operator<(const CacheKey& key) const {
+            return tableId < key.tableId ||
+                (tableId == key.tableId && keyHash < key.keyHash);
+        }
+    };
+
+    /**
+     * The Commit Cache is used to keep track of the  transaction operations to
+     * be performed during commit and well as cache read and write values to
+     * services subsequent reads.
+     */
+    typedef std::multimap<CacheKey, CacheEntry> CommitCacheMap;
+    CommitCacheMap commitCache;
+
+    /// Used to keep track of which cache entry to process next as part of the
+    /// commit protocol.
+    CommitCacheMap::iterator nextCacheEntry;
+
+    void initTask();
+    void processDecisionRpcs();
+    void processPrepareRpcs();
+    void sendDecisionRpc();
+    void sendPrepareRpc();
+
+    /// Encapsulates the state of a single Decision RPC sent to a single server.
+    class DecisionRpc : public RpcWrapper {
+        friend class ClientTransactionTask;
+      public:
+        DecisionRpc(RamCloud* ramcloud, Transport::SessionRef session,
+                    ClientTransactionTask* task);
+        ~DecisionRpc() {}
+
+        bool checkStatus();
+        bool handleTransportError();
+        void send();
+
+        void appendOp(CommitCacheMap::iterator opEntry);
+        void retryRequest();
+
+        /// Overall client state information.
+        RamCloud* ramcloud;
+
+        /// ClientTransactionTask that issued this rpc.
+        ClientTransactionTask* task;
+
+        /// Information about all of the ops that are being requested
+        /// in this RPC.
+#ifdef TESTING
+        static const uint32_t MAX_OBJECTS_PER_RPC = 3;
+#else
+        static const uint32_t MAX_OBJECTS_PER_RPC = 75;
+#endif
+        CommitCacheMap::iterator ops[MAX_OBJECTS_PER_RPC];
+
+        /// Header for the RPC (used to update count as objects are added).
+        WireFormat::TxDecision::Request* reqHdr;
+
+        DISALLOW_COPY_AND_ASSIGN(DecisionRpc);
+    };
+
+    /// Encapsulates the state of a single Prepare RPC sent to a single server.
+    class PrepareRpc : public RpcWrapper {
+        friend class ClientTransactionTask;
+      public:
+        PrepareRpc(RamCloud* ramcloud, Transport::SessionRef session,
+                ClientTransactionTask* task);
+        ~PrepareRpc() {}
+
+        bool checkStatus();
+        bool handleTransportError();
+        void send();
+
+        void appendOp(CommitCacheMap::iterator opEntry);
+        void retryRequest();
+
+        /// Overall client state information.
+        RamCloud* ramcloud;
+
+        /// ClientTransactionTask that issued this rpc.
+        ClientTransactionTask* task;
+
+        /// Information about all of the ops that are being requested
+        /// in this RPC.
```

Comment:
OK, but what exactly is that information? I.e., what is the meaning of an entry in ops?